### DISCUSSION TITLE
SizeVec?

### DISCUSSION THREAD

**Discussion by sarkahn** [bitshifter/glam-rs#261] [Ideas]
I was wondering if there's any chance we could get a "SizeVec" or "USizeVec" (Vec whose elements are usize). I use (usize,usize) a lot to represent 2D size, I find myself wanting to replace them with UVec2 to make the math easier to read, but then I have to convert to usize to interact with array indices and such.

**Comment by bitshifter** [comment]
Hey there. I am a bit reluctant to add this one. Since glam is not a generic library there's a bit more overhead to adding and maintaining new types. I will sit on this and see if there is further demand for it though.

If there is anything you can think of that would make it easier to work with usize with existing types let me know.

**Comment by sarkahn** [comment]
That makes sense, thanks for the consideration! I think even implementing `From<(usize,usize)>` and vice verse for UVec2 and UVec3 would make things a lot smoother for how I'm using them, if that would be possible.

**Comment by bitshifter** [comment]
I wouldn't add an `impl From` like this on the basis that Rust itself doesn't support `impl From<usize> for u32` or any other conversion between types that loses information. I could look at adding explicit methods, such as `from_tuple_usize` or something. You could easily add something similar in your own project as an extension method for `UVec*` .

**Comment by rayhem** [comment]
Just swinging by to +1 this. I'd find a `glam::USizeVec2` useful for exactly the same reason @sarkahn would: storing dimensions and interacting with array indices.

**Comment by amigash** [comment]
Same use case for me. It would be nice to have usizes available for 2d indexing.

**Comment by bitshifter** [comment]
Created an issue for this #578

**Comment by bitshifter** [comment]
Added
